data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java

```diff
@@ -13,68 +13,117 @@
 
 package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
 
+import static javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST;
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
 import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.compute_start_slot_at_epoch;
 
-import com.google.common.collect.ImmutableMap;
 import com.google.common.primitives.UnsignedLong;
-import java.util.List;
-import java.util.Map;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
 import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.tuweni.bytes.Bytes32;
-import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler;
-import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.beaconrestapi.schema.BadRequest;
+import tech.pegasys.artemis.beaconrestapi.schema.BeaconBlockResponse;
 import tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.artemis.provider.JsonProvider;
 import tech.pegasys.artemis.storage.ChainStorageClient;
 import tech.pegasys.artemis.storage.HistoricalChainData;
 
-public class BeaconBlockHandler implements BeaconRestApiHandler {
+public class BeaconBlockHandler implements Handler {
 
   private final ChainStorageClient client;
   private final HistoricalChainData historicalChainData;
+  public static final String ROUTE = "/beacon/block";
+  private final JsonProvider jsonProvider;
 
-  public BeaconBlockHandler(ChainStorageClient client, HistoricalChainData historicalChainData) {
+  public BeaconBlockHandler(
+      final ChainStorageClient client,
+      final HistoricalChainData historicalChainData,
+      final JsonProvider jsonProvider) {
     this.client = client;
     this.historicalChainData = historicalChainData;
+    this.jsonProvider = jsonProvider;
   }
 
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Request that the node return a specified beacon chain block.",
+      tags = {"Beacon"},
+      queryParams = {
+        @OpenApiParam(name = "epoch", description = "Query by epoch number (uint64)"),
```

Comment:
add a string constant for these, then we can reference them from tests and from here. something like EPOCH_PARAMETER